core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -1217,7 +1217,8 @@ public static IndexMetaData fromXContent(XContentParser parser) throws IOExcepti
      * {@link #isIndexUsingShadowReplicas(org.elasticsearch.common.settings.Settings)}.
      */
     public static boolean isOnSharedFilesystem(Settings settings) {
-        return settings.getAsBoolean(SETTING_SHARED_FILESYSTEM, isIndexUsingShadowReplicas(settings));
+        Version version = settings.getAsVersion(SETTING_VERSION_CREATED, Version.CURRENT);
```

Comment:
Are you sure it is a good idea to have a default here? This shouldn't be missing in production, right?